### PR TITLE
sway-scratch: init at 0.2.1

### PR DESCRIPTION
**Pull request by LilleAila**
## Description of changes

Package [aokellermann/sway-scratch](https://github.com/aokellermann/sway-scratch).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by IldenH** [comment]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [ ] package not built
- [ ] executable not tested
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [ ] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] `meta.mainProgram` is set, if applicable.
- [x] source is fetched using the appropriate function

##### Possible improvements
Add [automatic package updates](https://github.com/NixOS/nixpkgs/blob/1dcb998b7d19593c68ee3acd060d70ee7afe5367/pkgs/README.md#automatic-package-updates)
```nix
passthru.updateScript = nix-update-script { };
```

##### Comments

**Comment by LilleAila** [comment]
> ##### Reviewed points
> 
>     * [x]  package path fits guidelines
> 
>     * [x]  package name fits guidelines
> 
>     * [x]  package version fits guidelines
> 
>     * [ ]  package built
> 
>     * [ ]  executable tested
> 
>     * [x]  `meta.description` is set and fits guidelines
> 
>     * [x]  `meta.license` fits upstream license
> 
>     * [ ]  `meta.platforms` is set
> 
>     * [x]  `meta.maintainers` is set
> 
>     * [x]  `meta.mainProgram` is set, if applicable.
> 
>     * [x]  source is fetched using the appropriate function
> 
> 
> ##### Possible improvements
> 
> Add [automatic package updates](https://github.com/NixOS/nixpkgs/blob/1dcb998b7d19593c68ee3acd060d70ee7afe5367/pkgs/README.md#automatic-package-updates)
> 
> ```nix
> passthru.updateScript = nix-update-script { };
> ```
> 
> ##### Comments

Update script has been added, although i'm not sure whether or not it works with `cargoHash`.

**Comment by IldenH** [comment]
It does: https://github.com/Mic92/nix-update

> # Features
> - update buildRustPackage's cargoHash/cargoSha256/cargoLock and cargoSetupHook's cargoDeps

**Review by IldenH — APPROVED** [review]

**Comment by pbsds** [comment]
please mark broken on darwin or set `meta.platforms` to `linux`

**Comment by LilleAila** [comment]
> please mark broken on darwin or set `meta.platforms` to `linux`

Fixed now in the latest commit

File: pkgs/by-name/sw/sway-scratch/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "sway-scratch";
+  version = "0.2.1";
+  src = fetchFromGitHub {
+    owner = "aokellermann";
+    repo = "sway-scratch";
+    rev = "87df238d13dc15ba5733962c55e363766888724e";
```
Comment (pbsds) [inline pkgs/by-name/sw/sway-scratch/package.nix:14]:
```suggestion
    rev = "refs/tags/v${version}";
```

please make sure to recompute the hash as well

Comment (LilleAila) [reply pkgs/by-name/sw/sway-scratch/package.nix]: fixed

File: pkgs/by-name/sw/sway-scratch/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "sway-scratch";
+  version = "0.2.1";
+  src = fetchFromGitHub {
+    owner = "aokellermann";
+    repo = "sway-scratch";
+    rev = "v${version}";
+    hash = "sha256-wfwCg51Pfj4IGqOYK2Y/4If7PzZv9JHXewVWQDgbPko=";
```
Comment (IldenH) [inline pkgs/by-name/sw/sway-scratch/package.nix:15]: Hash needs to follow revision.
`$ nix flake prefetch --json github:aokellermann/sway-scratch/v0.2.1`
```suggestion
    hash = "sha256-1N/33XtkEWamgQYNDyZgSSaaGD+2HtbseEpQgrAz3CU=";
```
Make sure to do the same for cargoHash.

File: pkgs/by-name/sw/sway-scratch/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "sway-scratch";
+  version = "0.2.1";
```
Comment (SuperSandro2000) [inline pkgs/by-name/sw/sway-scratch/package.nix:10]:
```suggestion
  version = "0.2.1";

```

**Comment by pbsds** [comment]
```
error: hash mismatch in fixed-output derivation '/nix/store/k23vhv4hz7p3qv3smg6m4j7inx9rd807-source.drv':
         specified: sha256-wfwCg51Pfj4IGqOYK2Y/4If7PzZv9JHXewVWQDgbPko=
            got:    sha256-1N/33XtkEWamgQYNDyZgSSaaGD+2HtbseEpQgrAz3CU=
```

**Comment by LilleAila** [comment]
> ```
> error: hash mismatch in fixed-output derivation '/nix/store/k23vhv4hz7p3qv3smg6m4j7inx9rd807-source.drv':
>          specified: sha256-wfwCg51Pfj4IGqOYK2Y/4If7PzZv9JHXewVWQDgbPko=
>             got:    sha256-1N/33XtkEWamgQYNDyZgSSaaGD+2HtbseEpQgrAz3CU=
> ```

Don't know how that happened, but it should be fixed now

**Pull request merged**